pilot/pkg/serviceregistry/external/conversion.go

```diff
@@ -60,6 +61,26 @@ func convertServices(config model.Config) []*model.Service {
 	}
 
 	for _, host := range serviceEntry.Hosts {
+		var attrs model.ServiceAttributes
+		var meshExternal bool
+		switch serviceEntry.Location {
+		case networking.ServiceEntry_MESH_EXTERNAL:
+			meshExternal = true
+			attrs = model.ServiceAttributes{
+				Name:      host,
+				Namespace: config.Namespace,
+				ExportTo:  exportTo,
+			}
+		case networking.ServiceEntry_MESH_INTERNAL:
+			meshExternal = false
+			attrs = model.ServiceAttributes{
+				Name:      config.Name,
```

Comment:
how come this isn't `host` as well?

Reply:
I'm trying to replicate the logic in the k8s registry. TBH, I'm not entirely sure why the previous code was using host here and not the service name.  I'll be splitting this out into a separate PR eventually, so that I can have the right folks review it.